src/shogun/preprocessor/PCA.h

```diff
@@ -225,6 +225,10 @@ class CPCA: public CDimensionReductionPreprocessor
 		 * whitening to tackle numerical issues
 		 */
 		float64_t m_eigenvalue_zero_tolerance;
+
+	private:
+		void init_with_evd(const SGMatrix<float64_t>& feature_matrix, int32_t max_dim_allowed);
```

Comment:
These needs minimal doxygen documentation saying that they alter the state (transformation matrix).
Also potentially could say that EVD is based on an eigendocoposition of the covariance matrix, while SVD does an SVD on the data matrix.